1.24 fails to build on macOS 10.5.8: `listing.c: error: ‘const struct stat’ has no member named ‘st_birthtimespec’`

**Describe the bug**
Build fails on Leopard:
```
[ 45%] Building C object CMakeFiles/clifm.dir/src/listing.c.o
/usr/bin/gcc-4.2 -DCLIFM_DATADIR=/opt/local/share -I/opt/local/include -pipe -Os -DCLIFM_LEGACY -std=c99 -DNDEBUG -isystem/opt/local/include/LegacySupport -I/opt/local/include -arch ppc64 -mmacosx-version-min=10.5 -MD -MT CMakeFiles/clifm.dir/src/listing.c.o -MF CMakeFiles/clifm.dir/src/listing.c.o.d -o CMakeFiles/clifm.dir/src/listing.c.o -c /opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c: In function ‘set_long_attribs’:
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c:947: error: ‘const struct stat’ has no member named ‘st_birthtimespec’
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c: In function ‘compute_maxes’:
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c:1057: warning: comparison is always true due to limited range of data type
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c:1057: warning: comparison is always true due to limited range of data type
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c:1057: warning: comparison is always true due to limited range of data type
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c:1057: warning: comparison is always true due to limited range of data type
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c:1057: warning: comparison is always true due to limited range of data type
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c: In function ‘load_file_gral_info’:
/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/clifm-1.24/src/listing.c:2946: error: ‘const struct stat’ has no member named ‘st_birthtimespec’
make[2]: *** [CMakeFiles/clifm.dir/src/listing.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/build'
make[1]: *** [CMakeFiles/clifm.dir/all] Error 2
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_macos-powerpc.org_macos-powerpc_PPCPorts_ports_sysutils_clifm/clifm/work/build'
make: *** [all] Error 2
```

**To Reproduce**
Run the build on macOS 10.5.8.

**Expected behavior**
It would be nice if it compiles.

**Desktop (please complete the following information):**
 - OS: macOS
 - Clifm version: 1.24
- Installation source: Tarball from here via MacPorts.

**Additional context**
This was not found out earlier, because I only used 10.6 on powerpc, and there `st_birthtimespec` is supported, from what it looks, at least in 10.6.8.

--- a/src/helpers.h
+++ b/src/helpers.h
@@ -400,9 +400,13 @@ if (S_ISNWK(mode)) return 'n'; // HP/UX: network special file
  * actually store creation times: the value of __st_birthtim is always zero.
 #elif defined(__OpenBSD__)
 # define ST_BTIME __st_birthtim */
-# elif defined(__NetBSD__) || (defined(__APPLE__) && !defined(__ppc__) \
-&& !defined(__i386__))
+# elif defined(__NetBSD__)
 #  define ST_BTIME st_birthtimespec
+# elif defined(__APPLE__)
+#  include <AvailabilityMacros.h>
+#  if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
+#   define ST_BTIME st_birthtimespec
+#  endif
 # elif defined(__FreeBSD__) || defined(__CYGWIN__)
 #  define ST_BTIME st_birthtim
 # elif defined(__HAIKU__)

